src/librustdoc/html/render.rs

```diff
@@ -253,16 +253,26 @@ struct IndexItem {
 /// A type used for the search index.
 struct Type {
     name: Option<String>,
+    // true both for Option<T> and T, false otherwise.
+    generic: bool,
+    ty_params: Box<Vec<Type>>,
```

Comment:
Why `Box<Vec<..>>`? This is only useful in rare cases, Vec itself is already only three usize large.